### PR TITLE
Fix visibility in pages & components

### PR DESCRIPTION
**Pull request by florlafuente**
Fix visibility of pages and components according to user auth and user role.
Pages checked:

- index

- admin

- post

- profile

- auth

Please, @guillecro pay special attention to the small changes I made in `vote-like-reaction.js` cause the component coudn't be rendered if user was not logged in. Hope everything is working fine now :)

### CODE REVIEW

**Comment by guillecro** [comment]
> Please, @guillecro pay special attention to the small changes I made in vote-like-reaction.js cause the component coudn't be rendered if user was not logged in. Hope everything is working fine now :)

That was haunting me lately :cry: ... I dunno why sometimes it worked and sometimes it didn't
Flor teach me how to JS properly!

File: client/site/components/reactions/vote-like-reaction.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ export default class extends React.Component {
         this.setState({ myVote: vote })
       }
     } else {
-      this.setState({ myVote: {} })
+      this.setState({ myVote: null })
```
Comment (guillecro) [inline client/site/components/reactions/vote-like-reaction.js]: I cant believe this was the problem!! nngghh! / (╯°□°）╯︵ ┻━┻

File: pages/auth/callback.js [hunk 2]
```diff
@@ -4,10 +4,10 @@ import Router from 'next/router'
 import { NextAuth } from 'next-auth/client'
 
 export default class extends React.Component {
-  static async getInitialProps ({ req }) {
+  static async getInitialProps ({ req, query }) {
     return {
-      session: await NextAuth.init({ force: true, req: req }),
-      action: req.query.action
+      action: query.action,
+      session: await NextAuth.init({ force: true, req: req })
```
Comment (guillecro) [inline pages/auth/callback.js]: Whats whit this Flor? Would you like to explain me later?

Comment (florlafuente) [reply pages/auth/callback.js]: @guillecro It was very unstable and sometimes couldn't access to the query object. I checked next docs and the proper way to access the query object is via query and not req.query.

**Pull request merged**